core/arch/arm/mm/core_mmu.c

```diff
@@ -387,11 +465,15 @@ void core_mmu_set_discovered_nsec_ddr(struct core_mmu_phys_mem *start,
 	 * isn't aware of which memory range in the
 	 * non-secure memory is used for NSEC_SHM.
 	 */
-
 #ifdef CFG_SECURE_DATA_PATH
-	for (pmem = phys_sdp_mem_begin; pmem < phys_sdp_mem_end; pmem++)
-		carve_out_phys_mem(&m, &num_elems, pmem->addr, pmem->size);
-#endif
+	if (!configure_sdp_dtb(&m, &num_elems)) {
+		for (pmem = phys_sdp_mem_begin; pmem < phys_sdp_mem_end; pmem++)
+			carve_out_phys_mem(&m,
+					   &num_elems,
```

Comment:
I suppose it's possible to fold up a few parameters here.

Reply:
done here 2e3ba1b18cd2138acba8071f1617fc18a5ac0329